### ISSUE TITLE
Vertical stepper line is interrupted by its content

### ISSUE THREAD

**Issue by Pascalpex** [flutter/flutter#160156]
### Steps to reproduce

1. Run code example from https://api.flutter.dev/flutter/material/Stepper-class.html on 3.27.0
2. Observe interrupted line

### Expected results

The line between the first and the second step is continuous

### Actual results

The line between the first and the second step is interrupted

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

/// Flutter code sample for [Stepper].

void main() => runApp(const StepperExampleApp());

class StepperExampleApp extends StatelessWidget {
  const StepperExampleApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: const Text('Stepper Sample')),
        body: const Center(
          child: StepperExample(),
        ),
      ),
    );
  }
}

class StepperExample extends StatefulWidget {
  const StepperExample({super.key});

  @override
  State<StepperExample> createState() => _StepperExampleState();
}

class _StepperExampleState extends State<StepperExample> {
  int _index = 0;

  @override
  Widget build(BuildContext context) {
    return Stepper(
      currentStep: _index,
      onStepCancel: () {
        if (_index > 0) {
          setState(() {
            _index -= 1;
          });
        }
      },
      onStepContinue: () {
        if (_index <= 0) {
          setState(() {
            _index += 1;
          });
        }
      },
      onStepTapped: (int index) {
        setState(() {
          _index = index;
        });
      },
      steps: <Step>[
        Step(
          title: const Text('Step 1 title'),
          content: Container(
            alignment: Alignment.centerLeft,
            child: const Text('Content for Step 1'),
          ),
        ),
        const Step(
          title: Text('Step 2 title'),
          content: Text('Content for Step 2'),
        ),
      ],
    );
  }
}

```

</details>

The code is a copy from the example at https://api.flutter.dev/flutter/material/Stepper-class.html

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

Expected (Using 3.24.5):
![image](https://github.com/user-attachments/assets/4a06d4f5-d5d2-4006-8fda-10c45a677572)

Actual (Using 3.27.0):
![image](https://github.com/user-attachments/assets/c42b61d4-66b8-4ab2-af5f-eba1ecd7cc41)

</details>

### Logs

<details open><summary>Logs</summary>

```console
Not relevant
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[√] Flutter (Channel stable, 3.27.0, on Microsoft Windows [Version 10.0.22621.4460], locale de-DE)
    • Flutter version 3.27.0 on channel stable at C:\Users\pascalpex\dev\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 8495dee1fd (2 days ago), 2024-12-10 14:23:39 -0800
    • Engine revision 83bacfc525
    • Dart version 3.6.0
    • DevTools version 2.40.2

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at C:\Users\pascalpex\AppData\Local\Android\Sdk
    • Platform android-35, build-tools 35.0.0
    • ANDROID_HOME = C:\Users\pascalpex\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Java\jdk-17\bin\java
    • Java version Java(TM) SE Runtime Environment (build 17.0.12+8-LTS-286)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.12.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.12.35527.113
    • Windows 10 SDK version 10.0.22621.0

[√] Android Studio (version 2024.2)
    • Android Studio at C:\Users\pascalpex\AppData\Local\Programs\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-12282718-b509.11)

[√] IntelliJ IDEA Community Edition (version 2024.3)
    • IntelliJ at C:\Users\pascalpex\AppData\Local\Programs\IntelliJ IDEA Community Edition
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] VS Code (version 1.96.0)
    • VS Code at C:\Users\pascalpex\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.102.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.22621.4460]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 131.0.6778.140
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 131.0.2903.86

[√] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by Pascalpex** [comment]
This is a regression caused by https://github.com/flutter/flutter/pull/153619

**Comment by darshankawar** [comment]
Thanks for the report. Seeing the same behavior on latest stable and master versions, but not in previous stable (3.24.4).

**Comment by TahaTesser** [comment]
> This is a regression caused by #153619

cc: @nate-thegrate

**Comment by nate-thegrate** [comment]
> > This is a regression caused by #153619
> 
> cc: @nate-thegrate

Yep, that's on me, sorry! Will make sure it's fixed.

**Comment by siamahnaf** [comment]
What is the solutions for this problem. Should we wait for next version?

**Comment by siamahnaf** [comment]
@Pascalpex, how you solve this. Please help me?

**Comment by siamahnaf** [comment]
I am using 3.27.1 but still same issues?

**Comment by siamahnaf** [comment]
@darshankawar, Can you please help me to figure out how can I solve this issues. This issue is closed, but I am not finding any suggested solutions. I am using flutter version 3.27.1!

**Comment by Pascalpex** [comment]
> @Pascalpex, how you solve this. Please help me?

@siamahnaf 
The issue is solved on the master channel but not on in 3.27.1 or any other stable version. We will have to wait for the next major version or a potential hotfix in the future.
I have not solved this issue with any workaround other than using 3.24.5 or the master channel.
